java/src/jmri/util/zeroconf/ZeroConfClient.java

```diff
@@ -118,7 +119,7 @@ public ServiceInfo getServicebyAdName(String service, String adName) {
                 hostlist.add(info.getServer());
             }
         }
-        return ((String[]) hostlist.toArray());
+        return ((String[]) hostlist.toArray(new String[hostlist.size()]));
```

Comment:
I only noticed it because I fixed a similar cast caught by FindBugs earlier today.